addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -305,14 +305,21 @@ def _import_invoice(self, journal, filename, tree, existing_invoice=None):
                 # (Windows or Linux style) and/or the name of the xml instead of the pdf.
                 # Get only the filename with a pdf extension.
                 name = attachment_name.text.split('\\')[-1].split('/')[-1].split('.')[0] + '.pdf'
-                attachments |= self.env['ir.attachment'].create({
+                attachment = self.env['ir.attachment'].create({
                     'name': name,
                     'res_id': invoice.id,
                     'res_model': 'account.move',
                     'datas': text + '=' * (len(text) % 3),  # Fix incorrect padding
                     'type': 'binary',
                     'mimetype': 'application/pdf',
                 })
+                # Upon receiving an email (containing an xml) with a configured alias to create invoice, the xml is
+                # set as the main_attachment. To be rendered in the form view, the pdf should be the main_attachment.
+                if invoice.message_main_attachment_id and \
+                        invoice.message_main_attachment_id.name.endswith('.xml') and \
+                        'pdf' not in invoice.message_main_attachment_id.mimetype:
+                    invoice.message_main_attachment_id = attachment.id
```

Comment:
```suggestion
                    invoice.message_main_attachment_id = attachment
```
Sorry I didn't see this earlier